src/plugins/kibana/public/dashboard/services/saved_dashboards.js

```diff
@@ -41,6 +48,20 @@ define(function (require) {
       });
     };
 
+    this.scanAll = function (queryString, pageSize = 1000) {
+      return scanner.scanAndMap(queryString, {
+        pageSize,
+        docCount: Number.POSITIVE_INFINITY
+      }, this.mapHits.bind(this));
+    };
+
+    this.mapHits = function (hit) {
+      var source = hit._source;
+      source.id = hit._id;
+      source.url = this.urlFor(hit._id);
+      return source;
+    };
+
     this.find = function (searchString, size = 100) {
       var self = this;
```

Comment:
arrow functions could prevent the need for self, and should be used in place of the `fn.bind()` below.